apps/wappush/js/wappush.js

```diff
@@ -173,39 +173,46 @@
    * Send a notification for the provided message
    *
    * @param {Object} message The message the user needs to be notified of
+   *
+   * @return {Object} A promise that is resolved once the notification has been
+   *         sent and its events hooked up properly.
    */
   function wpm_sendNotification(message) {
-    var _ = navigator.mozL10n.get;
     var iconURL = NotificationHelper.getIconURI(app);
 
     /* Build the notification's text, for text/vnd.wap.connectivity-xml
      * messages this needs to be localized. */
-    var text = '';
-
-    if (message.text) {
-      text = (message.type == 'text/vnd.wap.connectivity-xml') ?
-             _(message.text) : message.text;
-    }
+    var body;
 
-    if (message.href) {
-      text += (text ? ' ' : '');
-      text += message.href;
+    if (message.type === 'text/vnd.wap.connectivity-xml') {
+      body = message.text; // The text will be localized
```

Comment:
Is there a chance that message.text will not have an assigned resource l10nId? Should we also have an array of localizable values and if message.text is from outside of that array do body = {raw: message.text} to follow https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/localization_code_best_practices#Writing_APIs_that_operate_on_L10nIDs ?